### PR TITLE
Events

### PR DESCRIPTION
**Pull request by Aboodchei**
screen shot 2019-01-30 at 8 53 29 pm
screen shot 2019-01-30 at 8 54 03 pm

### CODE REVIEW

File: app/models/job.rb [hunk 1]
```diff
@@ -1,8 +1,14 @@
 class Job < ApplicationRecord
-  belongs_to :organization
-  belongs_to :application_requirement
-  belongs_to :application_flow
-  belongs_to :department
-  has_many :links, as: :linkable
-  has_many :applications
+	include EventsTracker
+	track_creation
+	track_fields :application_requirement_id, :application_flow_id, :department,
```
Comment (sherif98) [inline app/models/job.rb]: should this be 
:department_id   ?
so it gets tracked as a relation?

File: lib/events_tracker.rb [hunk 2]
```diff
@@ -0,0 +1,71 @@
+module EventsTracker
```
Comment (sherif98) [inline lib/events_tracker.rb]: I think we should have some documentation here.
maybe include that the tracked model should have a title attribute and anything else you find relevant.

File: app/models/job.rb (hunk 1 above)
Comment (Aboodchei) [reply app/models/job.rb]: Yes, it should be `department_id`, thanks for the heads up.

File: lib/events_tracker.rb (hunk 2 above)
Comment (Aboodchei) [reply lib/events_tracker.rb]: Yeah, I was thinking that this is v1.0. This should keep us going fine for now, but the next version we will have to support different attributes than just :title, and we will also have to rethink the way the events are displayed. I was thinking that maybe every model would have it's own event display format. Let's meet later and brainstorm what we will need in the future for this.

**Pull request merged**